rasa/nlu/featurizers/dense_featurizer/mitie_featurizer.py

```diff
@@ -17,41 +20,94 @@
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # Specify what pooling operation should be used to calculate the vector of
+            # the complete utterance. Available options: 'mean' and 'max'
+            POOLING: MEAN_POOLING,
+        }
+
+    def __init__(
+        self,
+        config: Optional[Dict[Text, Any]],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> None:
+        super().__init__(config)
+        self.config = config
 
-        self.pooling_operation = self.component_config["pooling"]
+        # graph component
+        self._model_storage = model_storage
+        self._resource = resource
+        self._execution_context = execution_context
+
+        self.pooling_operation = self.config["pooling"]
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "MitieFeaturizerGraphComponent":
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, model_storage, resource, execution_context)
 
     @classmethod
-    def required_packages(cls) -> List[Text]:
+    def load(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+        **kwargs: Any,
+    ) -> "MitieFeaturizerGraphComponent":
+        """Loads trained component (see parent class for full docstring)."""
+        try:
+            loaded_featurizer = cls(
+                config=config,
+                model_storage=model_storage,
+                resource=resource,
+                execution_context=execution_context,
+            )
+            return loaded_featurizer
+        except ValueError:
+            logger.warning(
+                f"Failed to load {cls.__class__.__name__} from model storage. Resource "
+                f"'{resource.name}' doesn't exist."
+            )
+            return cls(config, model_storage, resource, execution_context)
+
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
         return ["mitie", "numpy"]
 
+    @classmethod
+    def required_components(cls) -> List[Type[Component]]:
+        return [MitieNLP, Tokenizer]
+
     def ndim(self, feature_extractor: "mitie.total_word_feature_extractor") -> int:
         return feature_extractor.num_dimensions
 
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
-
+    def train(self, training_data: TrainingData, **kwargs: Any,) -> None:
```

Comment:
This method should be renamed to `train_process` based on the issue description.